src/control/path_follower/path_follower/node_pure_pursuit.py

```diff
@@ -188,6 +214,9 @@ def callback(self, msg: PoseWithCovarianceStamped):
         error = wrap_to_pi(theta - des_heading_ang)
         steering_angle = np.rad2deg(error) * self.Kp_ang
 
+        # velocity control based on steering angle
+        des_vel2 = self.vel_min + max((self.vel_max - self.vel_min) * (1 - (abs(steering_angle) / 90) ** 2), 0)
+
         if self.DEBUG_IMG:
```

Comment:
This will need reworking when we merge particle pursuit - this is turned into a class function to remove duplicate code